hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -118,7 +130,30 @@ private BaselineEntity getBaselineEntity(
                 attrMap,
                 containerMap,
                 incomingEdges,
-                outgoingEdges));
+                outgoingEdges,
+                labelResultSet));
+  }
+
+  private LabelResultSet buildLabelResultSet(
+      org.hypertrace.gateway.service.v1.entity.Entity entity,
+      EntityRequest request,
+      Map<String, String> labelsMap) {
+    if (labelsMap.isEmpty()) {
```

Comment:
I don't think there is a need for this check anyways, since it would be automatically taken care of as we are streaming `labelsMap` on L151

Reply:
to return a static empty result set instead of creating an empty result set each time.